contrib/subrepo-cheri-libcxxrt/src/dwarf_eh.h

```diff
@@ -437,6 +437,10 @@ static bool dwarf_eh_find_callsite(struct _Unwind_Context *context,
 	result->landing_pad = 0;
 	// The current instruction pointer offset within the region
 	uint64_t ip = (char*)_Unwind_GetIP(context) - (char*)_Unwind_GetRegionStart(context);
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(__aarch64__)
+	// Account for the LSB in the address returned by _Unwind_GetIP().
+	ip--;
```

Comment:
```suggestion
	ip &= ~(uint64_t)1;
```
Should we use a bitwise and here in case the region start also happens to include the low bit (shouldn't ever happen, but seems slightly safer)? Alternatively we could have a debug assert that the low bit is set?

Reply:
If it does happen, then it wouldn't hurt because arm64 doesn't have single byte instructions.  If we really wanted to be paranoid, then we'd need something like:

```
#if defined(__CHERI_PURE_CAPABILITY__) && defined(__aarch64__)
    uint64_t ip = ((ptraddr_t)_Unwind_GetIP(context) & ~(ptraddr_t)1) - ((ptraddr_t)_Unwind_GetRegionStart(context) & ~(ptraddr_t)1);
#else
    // existing line
#endif
```

That just seems quite verbose.  Alternatively I could add helper variables for GetIP() and GetRegionStart():

```
    ptraddr_t ipAddr = (ptraddr_t)_Unwind_GetIP(context);
    ptraddr_t regionStart = (ptraddr_t)_Unwind_GetRegionStart(context);
#if defined(__CPC__) && defined(__aarch64__)
    ipAddr &= ~1;
    regionStart &= ~1;
#endif
    ip = ipAddr - regionStart;
```

This is also more verbose in terms of a diff to the existing code, but the shorter lines are perhaps more readable.